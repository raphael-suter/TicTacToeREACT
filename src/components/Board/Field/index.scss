@import "../../../styles/colors.scss";

.field {
  margin: 0.3rem;
  position: relative;

  &::before {
    content: "";
    display: block;
    padding-bottom: 100%;
  }

  button {
    position: absolute;
    top: 0;
    bottom: 0;
    width: 100%;
    font-size: 52px;
    background: $mediumColor;
    box-shadow: inset 0 0 2px 0 $shadow;
    border: none;
    cursor: pointer;
  }
}
