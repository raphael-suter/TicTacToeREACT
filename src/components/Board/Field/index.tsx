import React from 'react';
import './index.scss';
import Props from './Props';

export default ({ content, onClick }: Props) => (
  <div className='board-field'>
    <button onClick={onClick} aria-label='field'>{content}</button>
  </div>
);
