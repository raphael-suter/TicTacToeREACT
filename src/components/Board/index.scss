@import "../../styles/colors.scss";
@import "../../styles/mixins.scss";

.board {
  padding: 5% 1.2rem 1.2rem;

  .grid {
    @include card;
    display: flex;
    flex-wrap: wrap;
  }
}
