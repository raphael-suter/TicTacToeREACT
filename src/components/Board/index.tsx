import React from 'react';
import './index.scss';
import Props from './Props';

export default ({ children }: Props) => (
  /* Two nested boxes are necessary to make sure that there's always some spacing around the inner one. 
  (grid's margin is set to auto) */
  <main className='board'>
    <div className='board__grid'>
      {children}
    </div>
  </main>
);
