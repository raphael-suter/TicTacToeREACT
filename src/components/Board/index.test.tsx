import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Board from '.';

describe('Board', () => {
  test('Renders correctly.', () => {
    const wrapper = shallow(<Board children={null} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
