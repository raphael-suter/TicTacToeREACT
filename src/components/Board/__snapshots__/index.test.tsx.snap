// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Renders correctly. 1`] = `
<styled.main>
  <styled.div />
</styled.main>
`;
